### PR TITLE
Add ability to bind onChange to autocomplete hook

### PR DESCRIPTION
**Pull request by vincaslt**
proton-shared PR for identity function: https://github.com/ProtonMail/proton-shared/pull/24

Initial idea was that if you wanted to keep track of selected/removed events in components using autocomplete hook, you would just create custom functions like:

```
const handleSelect = (item, label) => {
  select(item, label);
  // your stuff
}
```

But this way you don't yet have access to updated selectedItems list, because it's going to be updated after the next render/state update, so you'd end upt needing to reimplement the whole select function from hook. Solved this by adding an effect, that will call onChange after selectedItems changes.

```
const {...} = useAutocomplete({
  onChange: (selectedItems) => {
    // your stuff using updated selectedItems
  }
})
```

### CODE REVIEW

File: components/autocomplete/useAutocomplete.js [hunk 1]
```diff
@@ -1,16 +1,29 @@
-import { useState } from 'react';
+import { useState, useEffect } from 'react';
+import { noop } from 'proton-shared/lib/helpers/function';
 
-const defaultItemToLabel = (label) => label;
+const defaultLabelToItem = (label) => label;
```
Comment (dhoko) [inline components/autocomplete/useAutocomplete.js]: Can you add the identity function inside proton-shared ?

File: components/autocomplete/useAutocomplete.test.js [hunk 2]
```diff
@@ -43,4 +43,19 @@ describe('useAutocomplete hook', () => {
...
+        const initialSelectedItems = [{ label: 'test', value: 'T' }];
+        const { result } = renderHook(() =>
+            useAutocomplete({
+                multiple: true,
+                initialSelectedItems,
+                onChange: onChangeSpy
+            })
+        );
+        act(() => result.current.select(newItem, newItem.label));
+        expect(onChangeSpy).toBeCalledWith([...initialSelectedItems, newItem]);
```
Comment (dhoko) [inline components/autocomplete/useAutocomplete.test.js]: You don't test that you don't call onChange during the init ;)

Comment (vincaslt) [reply components/autocomplete/useAutocomplete.test.js]: ah, good idea

File: components/autocomplete/useAutocomplete.js [hunk 3]
```diff
@@ -1,16 +1,29 @@
...
 
 const useAutocomplete = ({
     multiple = true,
     initialSelectedItems = [],
     initialInputValue = '',
-    labelToItem = defaultItemToLabel
+    labelToItem = defaultLabelToItem,
+    onChange = noop
 } = {}) => {
+    const [initialized, setInitialized] = useState(false);
```
Comment (mmso) [inline components/autocomplete/useAutocomplete.js]: If you don't need to trigger a re-render it'd be better to `useRef` instead

**Pull request merged**